### DISCUSSION TITLE
TDI Strategy based on Trading Made Strategy from EElfranz

### DISCUSSION THREAD

**Discussion by mavidelisi** [EA31337/EA31337#390] [Ideas]
Hello Kenorb ;
I am trying to create the Trading Made Simple Strategy on the 4Hr chart of Eelfranz from Forexfactory, but the TDI indicator is not integrated into the system, and at the same time, the heiken ashi strategy only opens a buying position. I have created a request for Heiken ashi to correct the error. In the same way, I created a request for the TDI indicator to be integrated into the system.  Can you help me?

> Trading Made Strategy from EElfranz (https://www.forexfactory.com/thread/291622-trading-made-simple)

Thank you.
![u-chf](https://github.com/EA31337/EA31337/assets/14947927/868a77ba-d223-44e9-b62d-78884b375e5c)

**Comment by kenorb** [comment]
Few side notes.
When writing new strategies, you don't need to rely other strategies logic. For example issue with Heiken Ashi only apply to other strategy, not indicator. You can still use Heiken Ashi to average candles. Plus, you don't need indicators part of the repositories (system), you can load them from the same repo/folder than is your new strategy.

I'm not sure the differences between _Trading Made Strategy_ vs E.A.S.Y. strategy (to which there are few PDFs which describe it quiet clar), so I've created a separate suggestion just in case: https://github.com/EA31337/EA31337-strategies/discussions/68

In general, it's a good idea to implement it.

So we need two indicators: HA (Heiken Ashi) and TDI. Plus Stochastics is mentioned as well, so 3 in total? Can you clarify?

Can you write summary and clarify on trading conditions?

I'll look into other issues which you've posted.

**Reply by mavidelisi** [reply]
yes, the same strategy. Only the heiken ashi problem is left. it would be nice if you could solve it too

**Comment by kenorb** [comment]
I've implemented new [TDI Strategy](https://github.com/EA31337/Strategy-TDI), although it's not complete yet, so let me know any issues after your tests. The idea of running indicator on Heiken Ashi's OHLC candles (instead of regular chart) isn't specific to this strategy, as any can benefit of it. It's something that is being implemented in the EA framework and can be used across all indicators and strategies. I'm going to check if the codebase is ready for that integration. Re adding Stochastics indicator, it'll need to be another strategy cloned from TDI (or maybe the same, we'll see), so it can benefit from 2 indicators at the same time. So let's first finalize TDI strategy, then we can add more indicators into it.
